pkg/operator/apis/config/v1alpha1/types.go

```diff
@@ -98,13 +99,70 @@ type GardenControllerConfig struct {
+	// Monitoring contains the configuration for the monitoring components of the runtime cluster.
+	// +optional
+	Monitoring *GardenMonitoring `json:"monitoring,omitempty"`
+}
+
+// PrometheusConfig contains configuration for Prometheus components in the Garden cluster.
+type PrometheusConfig struct {
+	// Retention is the retention size for the Prometheus data.
+	// +optional
+	Retention resource.Quantity `json:"retention,omitempty"`
+	// Storage is the storage configuration for Prometheus.
+	// +optional
+	Storage *Storage `json:"storage,omitempty"`
+}
+
+// Storage contains storage configuration.
+type Storage struct {
+	// Capacity is the storage capacity for the volumes.
+	// +kubebuilder:default=`10Gi`
+	// +optional
+	Capacity *resource.Quantity `json:"capacity,omitempty"`
+	// ClassName is the name of a storage class.
+	// +optional
+	ClassName *string `json:"className,omitempty"`
+}
+
+// GardenMonitoring contains the configuration for the monitoring components of the runtime cluster.
+type GardenMonitoring struct {
+	// PrometheusGarden contains the configuration for the Prometheus components in the Garden cluster.
+	// +optional
+	PrometheusGarden *PrometheusConfig `json:"prometheusGarden,omitempty"`
+	// PrometheusLongterm contains the configuration for the long-term Prometheus components in the Garden cluster.
+	// +optional
+	PrometheusLongterm *PrometheusConfig `json:"prometheusLongterm,omitempty"`
+}
+
+// SeedMonitoring contains the configuration for the monitoring components of the seed cluster.
+type SeedMonitoring struct {
+	// PrometheusCache contains the configuration for the Prometheus cache components in the seed cluster.
+	// +optional
+	PrometheusCache *PrometheusConfig `json:"prometheusCache,omitempty"`
+	// PrometheusSeed contains the configuration for the Prometheus components in the seed cluster.
+	// +optional
+	PrometheusSeed *PrometheusConfig `json:"prometheusSeed,omitempty"`
+	// PrometheusAggretate contains the configuration for the Prometheus aggregate components in the seed cluster.
+	// +optional
+	PrometheusAggretate *PrometheusConfig `json:"prometheusAggretate,omitempty"`
+}
+
+// GardenletMonitoring contains the configuration for the monitoring components of the gardenlet.
+type GardenletMonitoring struct {
+	// Seed contains the configuration for the monitoring components of the seed cluster.
+	// +optional
+	Seed SeedMonitoring `json:"seed,omitempty"`
 }
 
 // GardenletDeployerControllerConfig is the configuration for the gardenlet deployer controller.
 type GardenletDeployerControllerConfig struct {
 	// ConcurrentSyncs is the number of concurrent worker routines for this controller.
 	// +optional
 	ConcurrentSyncs *int `json:"concurrentSyncs,omitempty"`
+	// Monitoring contains the configuration for the monitoring components of the runtime cluster.
+	// +optional
+	Monitoring *GardenletMonitoring `json:"monitoring,omitempty"`
```

Comment:
Again, why is this needed? The monitoring for the gardenlet can be configured directly in the garden config, no need to configure `gardener-operator`'s controller for reconciling `Gardenlet` resources with it, or am I missing something?